app.js

```diff
@@ -2,17 +2,107 @@ require('dotenv').config();
 
 const express = require('express');
 const hbs = require('hbs');
+const SpotifyWebApi = require('spotify-web-api-node');
 
-// require spotify-web-api-node package here:
 
+// require spotify-web-api-node package here:
 const app = express();
 
 app.set('view engine', 'hbs');
 app.set('views', __dirname + '/views');
 app.use(express.static(__dirname + '/public'));
 
 // setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+    clientId: process.env.CLIENT_ID,
+    clientSecret: process.env.CLIENT_SECRET
+});
+
+spotifyApi
+    .clientCredentialsGrant()
+    .then(data => spotifyApi.setAccessToken(data.body['access_token']))
+    .catch(error => console.log('Something went wrong when retrieving an access token', error));
+
 
 // Our routes go here:
+app.get('/', (req, res) => {
+    res.render('index');
+});
+
+app.get('/artist-search', (req, res) => {
+
+    spotifyApi
+        .searchArtists(req.query.artist)
+        .then(data => {
+            const spotifyArray = data.body.artists.items;
+            // ----> 'HERE WHAT WE WANT TO DO AFTER RECEIVING THE DATA FROM THE API'
+            const artistArray = spotifyArray.map(function (element) {
+                return {
+                    name: element.name,
+                    image: element.images[0],
+                    id: element.id
+                };
+            });
+            res.render('artist-search-results', { artistArray });
+        })
+        .catch(err => console.log('The error while searching artists occurred: ', err));
+});
+
+app.get('/albums/:artistId', (req, res, next) => {
+
+    spotifyApi.getArtistAlbums(
+        req.params.artistId, {
+            limit: 10,
+            offset: 20
+        },
+        
+        function (err, data) {
+            
+            if (err) {
+                console.error('Something went wrong!', err);
+            } else {
+                const spotifyAlbumsArray = data.body.items;
+
+                const albumsArray = spotifyAlbumsArray.map(function (album) {
+                    return {
+                        name: album.name,
+                        image: album.images[0],
+                        id: album.id
+                    };
+                });
```

Comment:
I liked this approach, that you are organizing the data before sending it to the view.

Reply:
Thank you Jeff! There have been some moments of frustration involved. XD